chia/_tests/clvm/test_member_puzzles.py

```diff
@@ -291,6 +292,82 @@ async def test_passkey_member(cost_logger: CostLogger) -> None:
+        delegated_puzzle_hash = delegated_puzzle.get_tree_hash()
+
+        # setup keys
+        seed = 0x1A62C9636D1C9DB2E7D564D0C11603BF456AAD25AA7B12BDFD762B4E38E7EDC6
+        secp_sk = ec.derive_private_key(seed, ec.SECP256R1(), default_backend())
+        secp_pk = secp_sk.public_key().public_bytes(Encoding.X962, PublicFormat.CompressedPoint)
+
+        passkey_member = PasskeyPuzzleAssertMember(secp_pk, sim.defaults.GENESIS_CHALLENGE)
+
+        passkey_puzzle = PuzzleWithRestrictions(0, [], passkey_member)
+
+        # Farm and find coin
+        await sim.farm_block(passkey_puzzle.puzzle_hash())
+        coin = (
+            await client.get_coin_records_by_puzzle_hashes([passkey_puzzle.puzzle_hash()], include_spent_coins=False)
+        )[0].coin
+        block_height = sim.block_height
+
+        # Create an announcements to be asserted in the delegated puzzle
+        announcement = CreateCoinAnnouncement(msg=b"foo", coin_id=coin.name())
+
+        # Get signature for AGG_SIG_ME
+        authenticator_data = b"foo"
+        client_data = {"challenge": passkey_member.create_message(delegated_puzzle_hash, coin.puzzle_hash)}
+        client_data_hash = std_hash(PasskeyMember.format_client_data_as_str(client_data).encode("utf8"))
+        signature_message = authenticator_data + client_data_hash
+        der_sig = secp_sk.sign(
+            signature_message,
+            # The type stubs are weird here, `deterministic_signing` is assuredly an argument
+            ec.ECDSA(hashes.SHA256(), deterministic_signing=True),  # type: ignore[call-arg]
+        )
+        r, s = decode_dss_signature(der_sig)
+        sig = r.to_bytes(32, byteorder="big") + s.to_bytes(32, byteorder="big")
+        sb = WalletSpendBundle(
+            [
+                make_spend(
+                    coin,
+                    passkey_puzzle.puzzle_reveal(),
+                    passkey_puzzle.solve(
+                        [],
+                        [],
+                        passkey_member.solve(
+                            authenticator_data,
+                            client_data,
+                            sig,
+                            coin.puzzle_hash,
+                        ),
+                        DelegatedPuzzleAndSolution(
+                            delegated_puzzle,
+                            Program.to(
+                                [
+                                    announcement.to_program(),
+                                    announcement.corresponding_assertion().to_program(),
+                                ]
+                            ),
+                        ),
+                    ),
+                )
+            ],
+            G2Element(),
+        )
+        result = await client.push_tx(
+            cost_logger.add_cost(
+                "Passkey w/ puzzle assert spendbundle",
+                sb,
+            )
+        )
+        assert result == (MempoolInclusionStatus.SUCCESS, None)
```

Comment:
Can you add a test with an incorrect signature and a test with an incorrect `client_data_json`?

Reply:
Going to resolve in a separate PR